### PR TITLE
Runtime watcher update game room instance when receive added event

### PR DESCRIPTION
**Pull request by gabrielcorado**
Implementing the `Start`, `Stop`, and `IsRunning` of the runtime watcher. Implementing processing of `InstanceEventTypeAdded` event.

This PR also adds a new function to the `RoomManager` used to update the game room instance (`UpdateRoomInstance`).

### CODE REVIEW

File: internal/core/workers/runtime_watcher_worker/runtime_watcher_worker_test.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+		config := room_manager.RoomManagerConfig{RoomInitializationTimeoutMillis: time.Millisecond * 1000}
+		roomManager := room_manager.NewRoomManager(fakeClock, portAllocator, roomStorage, instanceStorage, runtime, config)
+
+		return mockCtrl, instanceStorage, runtime, &workers.WorkerOptions{
+			Runtime:     runtime,
+			RoomManager: roomManager,
+		}
+	}
+
+	t.Run("fails to start watcher", func(t *testing.T) {
```
Comment (GuilhermBrSp) [inline internal/core/workers/runtime_watcher_worker/runtime_watcher_worker_test.go:65]: @gabrielcorado WDYT about adding one test case for when `w.roomManager.UpdateRoomInstance` call fails?

**Review by GuilhermBrSp — APPROVED** [review]

File: internal/core/workers/runtime_watcher_worker/runtime_watcher_worker_test.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+		mockCtrl, _, runtime, workerOptions := workerOptions(t)
```
Comment (GuilhermBrSp) [inline internal/core/workers/runtime_watcher_worker/runtime_watcher_worker_test.go:66]: Also, is there some way that we can test the `IsRunning` method?

File: internal/core/workers/runtime_watcher_worker/runtime_watcher_worker.go [hunk 3]
```diff
@@ -24,29 +24,84 @@ package runtime_watcher_worker
...
 var _ workers.Worker = (*runtimeWatcherWorker)(nil)
 
 // runtimeWatcherWorker is a work that will watch for changes on the Runtime
 // and apply to keep the Maestro state up-to-date. It is not expected to perform
 // any action over the game rooms or the scheduler.
-type runtimeWatcherWorker struct{}
+type runtimeWatcherWorker struct {
+	scheduler   *entities.Scheduler
+	roomManager *room_manager.RoomManager
+	// TODO(gabrielcorado): should we access the port directly? do we need to
```
Comment (GuilhermBrSp) [inline internal/core/workers/runtime_watcher_worker/runtime_watcher_worker.go:45]: @gabrielcorado  I didn't get your point here

File: internal/core/workers/runtime_watcher_worker/runtime_watcher_worker_test.go (hunk 1 above)
Comment (gabrielcorado) [reply internal/core/workers/runtime_watcher_worker/runtime_watcher_worker_test.go:65]: Sure.

File: internal/core/workers/runtime_watcher_worker/runtime_watcher_worker_test.go (hunk 2 above)
Comment (gabrielcorado) [reply internal/core/workers/runtime_watcher_worker/runtime_watcher_worker_test.go:66]: I've added a check on the current tests.

File: internal/core/workers/runtime_watcher_worker/runtime_watcher_worker.go (hunk 3 above)
Comment (gabrielcorado) [reply internal/core/workers/runtime_watcher_worker/runtime_watcher_worker.go:45]: Until now, we're only accessing ports from core services. The workers are not services, so should we add this "interface" to some of our services?

**Pull request merged**